### PR TITLE
Brief guide for deploying

### PR DESCRIPTION
**Pull request by zspencer**
**Why?** - Deploying is scary! Lists make it less scary. Automation makes it even less scarier; but I didn't want to dive too deep too fast.

### CODE REVIEW

File: DEPLOYING.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+### Timing for Deploys
+
+* Staging: Deployed on successful builds from Master.
+* Production: Deployed after acceptance at standup.
+
+### Pre-Production Checklist
+- [ ] Review the buildpacks on staging and production. Are they the same? If
+  not, make them the same.
+- [ ] Skim the diff between the current production sha and master
+  - `git fetch --all && git diff production/master staging/master`
```
Comment (jessieay) [inline DEPLOYING.md]: maybe provide github URL as well for those who are more comfortable viewing diffs there?

File: DEPLOYING.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  - [ ] Any new ENV variables? If so, set them.
+  - [ ] What about rake tasks? If so, make note of them to run after deploy.
```
Comment (jessieay) [inline DEPLOYING.md]: this would be for a one-off, right?

File: DEPLOYING.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  - [ ] Any scary looking migrations that could result in data loss or prevent
```
Comment (jessieay) [inline DEPLOYING.md]: seems like this should be a part of code review

File: DEPLOYING.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    roll back? If so, perhaps we should hold off on the deploy until we fix the
+    scary migrations
+
+### Actually Deploying Production
+- [ ] `heroku pipelines:promote -a michigan-benefits-staging`
+- [ ] Run post-deploy rake tasks
```
Comment (jessieay) [inline DEPLOYING.md]: maybe add a step for clicking around on prod to ensure it's not broken ?

File: bin/setup [hunk 5]
```diff
@@ -7,6 +7,11 @@ chdir APP_ROOT do
   # This script is a starting point to setup your application.
   # Add necessary setup steps to this file.
 
+  step "Adding heroku remotes for `-r` happinesss" do
+    system!("heroku git:remote -a michigan-benefits-staging -r staging")
+    system!("heroku git:remote -a michigan-benefits-production -r production")
```
Comment (jessieay) [inline bin/setup]: yay!

File: DEPLOYING.md (hunk 2 above)
Comment (zspencer) [reply DEPLOYING.md]: Yep!

File: DEPLOYING.md (hunk 1 above)
Comment (zspencer) [reply DEPLOYING.md]: like, as a template for them to fill out or as a script to run that actually figures out what the URL should be? (I think both are valid)

File: DEPLOYING.md (hunk 3 above)
Comment (zspencer) [reply DEPLOYING.md]: Yeah, I would hope that to be the case.

File: DEPLOYING.md (hunk 4 above)
Comment (zspencer) [reply DEPLOYING.md]: Yah, I am also comfortable putting in a faux-office that faxes to our staging environment as part of this PR. Thoughts?

File: DEPLOYING.md (hunk 1 above)
Comment (jessieay) [reply DEPLOYING.md]: either works for me!

File: DEPLOYING.md (hunk 4 above)
Comment (jessieay) [reply DEPLOYING.md]: sure!

File: DEPLOYING.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+### Timing for Deploys
+
+* Staging: Deployed on successful builds from Master.
```
Comment (adarsh) [inline DEPLOYING.md]: Maybe "Deployed _automatically_ on successful builds from Master"?

**Comment by adarsh** [comment]
What do you all think about a `bin/deploy` script which encapsulates whatever details we have? It wouldn't cover things like "Hey, did you add some `ENV` references?" but it's something.

I found this one from a previous project:

```
#!/bin/sh

# Run this script to deploy the app to Heroku.
#
# ./bin/deploy (pushes to staging by default)
# ./bin/deploy staging
# ./bin/deploy production

set -e

branch="$(git symbolic-ref HEAD --short)"
target="${1:-staging}"

git push "$target" "$branch:master"
heroku run rake db:migrate --exit-code --remote "$target"
heroku restart --remote "$target"
```

**Comment by jayroh** [comment]
> What do you all think about a bin/deploy script which encapsulates whatever details we have? 

Big fan of this. Deploys end up incrementally growing into a non-trivial process so this ends up being extremely helpful.

File: bin/release-diff [hunk 7]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -e
+
+git fetch --all
+
+staging_sha=$(git log --oneline staging/master | head -n 1 | cut -d " " -f1)
+prod_sha=$(git log --oneline production/master | head -n 1 | cut -d " " -f1)
+
+open "https://github.com/codeforamerica/michigan-benefits/compare/$prod_sha...$staging_sha"
```
Comment (jessieay) [inline bin/release-diff]: beauty!

**Pull request merged**